File: Tareas Java/Operaciones con Arreglos Csharp/Operaciones con Arreglos Csharp/Forms/FormRecepcionDatos.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Operaciones_con_Arreglos_Csharp.Forms;

namespace Operaciones_con_Arreglos_Csharp.Forms
{
    public partial class FormRecepcionDatos : Form
    {
        public FormRecepcionDatos()
        {
            InitializeComponent();
        }


        public int cantidad;
        public double[] arreglo;
        int i = 1;
        FormResultado formularioResultado = new FormResultado();

        private void btnSiguiente_Click(object sender, EventArgs e)
        {
            arreglo[i - 1] = Convert.ToDouble(txtNumero.Text);

            //Actualizamos el label y el textbox
            label1.Text = "Elemento " + Convert.ToString(i + 1);
            txtNumero.Text = "";

            if (i == cantidad)
            {
                formularioResultado.arreglo = new double[cantidad];
                Array.Copy(arreglo, formularioResultado.arreglo, cantidad);
                formularioResultado.cantidad = cantidad;

                this.Hide();
                formularioResultado.Show();
            }
            else {
                i = i + 1;
            }

        }

        private void FormRecepcionDatos_Load(object sender, EventArgs e)
        {
            arreglo = new double[cantidad];
        }

    }
}

File: Tareas Java/Operaciones con Arreglos vb/Operaciones con Arreglos vb/FormRecepcionDatos.vb
Public Class FormRecepcionDatos

    Public cantidad As Integer
    Public arreglo() As Double
    Dim i As Integer = 1

    Dim formularioResultado As New FormResultado()

    Private Sub btnSiguiente_Click(sender As Object, e As EventArgs) Handles btnSiguiente.Click

        arreglo(i - 1) = Convert.ToDouble(txtNumero.Text)

        'Actualizamos el label y el textbox
        Label1.Text = "Elemento " + Convert.ToString(i + 1)
        txtNumero.Text = ""

        Select Case i
            Case cantidad

                ReDim formularioResultado.arreglo(cantidad - 1)
                Array.Copy(arreglo, formularioResultado.arreglo, cantidad)
                formularioResultado.cantidad = cantidad

                Me.Hide()
                formularioResultado.Show()

            Case Else

                i = i + 1
        End Select




    End Sub

    Private Sub FormRecepcionDatos_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ReDim arreglo(cantidad - 1)

    End Sub
End Class